workflow/templateresolution/context.go

```diff
@@ -166,6 +199,17 @@ func (ctx *Context) GetTemplateScope() string {
 	return string(ctx.tmplBase.GetResourceScope()) + "/" + ctx.tmplBase.GetName()
 }
 
+// UpdateTemplateStatus update the WorkflowTemplate.Status of a given WorkflowTemplate.
+func (ctx *Context) UpdateTemplateStatus(wftmple *wfv1.WorkflowTemplate) (*wfv1.WorkflowTemplate, error) {
```

Comment:
UpdateTemplateStatus should not be capitalised, as it is not necessary to [export](https://go.dev/tour/basics/3) it. Rename to `updateTemplateStatus`.